Sources/MapboxMaps/Style/StyleURI.swift

```diff
@@ -2,91 +2,39 @@ import Foundation
 
 /// Enum representing the latest version of the Mapbox styles (as of publication). In addition,
 /// you can provide a custom URL or earlier version of a Mapbox style by using the `.custom` case.
-public enum StyleURI: Hashable, RawRepresentable {
-    public typealias RawValue = URL
+public struct StyleURI: Hashable, RawRepresentable {
+    public typealias RawValue = String
+
+    public let rawValue: String
+
+    /// Create a custom StyleURI from a URL. The URL may be a full HTTP or HTTPS URL, a Mapbox style URL
+    /// (mapbox://styles/{user}/{style}), or a path to a local file relative to the application’s
+    /// resource path.
+    /// Returns nil if the URI is invalid.
+    /// - Parameter rawValue: URL for the style
+    public init?(rawValue: String) {
```

Comment:
Should the raw value be string or URL? Making it URL would guarantee well-formedness at the earliest possible opportunity, so that we wouldn’t need to manually validate the raw value before using it. Either way, consider adding a URL-typed convenience initializer to head off developer mistakes like using `path` instead of `absoluteString`.

Reply:
> Should the raw value be string or URL?

I switched to String because that's what we ultimately pass to GL Native.

> Making it URL would guarantee well-formedness at the earliest possible opportunity, so that we wouldn’t need to manually validate the raw value before using it.

I believe the next line provides the safety you're looking for:

`guard let url = URL(string: rawValue), url.scheme != nil else`

Even when the raw value was a URL, we were still validating that its scheme was non-nil.

> Either way, consider adding a URL-typed convenience initializer to head off developer mistakes like using path instead of absoluteString.

I'm on board with this.